gobblin-api/src/test/java/org/apache/gobblin/util/io/GsonInterfaceAdapterTest.java

```diff
@@ -43,4 +44,21 @@ public void test() {
 
   }
 
+  @Test
+  public void testObjectToIntegerDeserialize() {
+    Gson gson = new GsonBuilder().create();
+    Integer test = 5;
```

Comment:
please add test for other scenarios i.e. long, floating point numbers and invalid string